src/aeso_ast_infer_types.erl

```diff
@@ -2185,6 +2250,9 @@ unify1(_Env, {qcon, _, Name}, {qcon, _, Name}, _When) ->
     true;
 unify1(_Env, {bytes_t, _, Len}, {bytes_t, _, Len}, _When) ->
     true;
+unify1(Env, {if_t, _, {id, _, Id}, Then1, Else1}, {if_t, _, {id, _, Id}, Then2, Else2}, When) ->
+    unify(Env, Then1, Then2, When) andalso
+    unify(Env, Else1, Else2, When);
```

Comment:
Why doesn't `{if_t, {bool, _, true}, T, _}` unify with `T`?

Reply:
The argument to `if_t` is always an `id()`. As soon as we know the value the `if_t` goes away.